onnxruntime/python/onnxruntime_inference_collection.py

```diff
@@ -359,11 +359,8 @@ def __init__(
         """
         super().__init__()
 
-        if isinstance(path_or_bytes, str):
-            self._model_path = path_or_bytes
-            self._model_bytes = None
-        elif isinstance(path_or_bytes, os.PathLike):
-            self._model_path = str(path_or_bytes)
+        if isinstance(path_or_bytes, (str, os.PathLike)):
+            self._model_path = os.fspath(path_or_bytes)
```

Comment:
I thought os.fsdecode is preferred than os.fspath? os.fspath could return bytes. os.fsdecode convert such things to strings.

Reply:
We don't handle bytes inputs in this case so that's ok?

Reply:
Or should I use `os.path.abspath`?

Reply:
I'm not familiar with these python APIs, cannot make a good suggestion.

Reply:
I this case I will follow what Brett says in https://github.com/charliermarsh/ruff/issues/3675#issuecomment-1494975508 and keep fspath. Any case it is better than `str` that I used